core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt

```diff
@@ -378,9 +387,42 @@ internal class CBPeripheralCoreBluetoothPeripheral(
         }
 
         val platformDescriptor = discoveredServices.obtain(descriptor)
-        return connection.execute<DidUpdateValueForDescriptor> {
+        val updatedDescriptor = connection.execute<DidUpdateValueForDescriptor> {
             centralManager.read(cbPeripheral, platformDescriptor)
-        }.descriptor.value as NSData
+        }.descriptor
+
+        // CBDescriptor "value" property type depends on cbDescriptor.UUID.UUIDString
+        // https://stackoverflow.com/questions/52816049/how-to-convert-cbdescriptor-value-to-string
+        // https://developer.apple.com/documentation/corebluetooth/characteristic-descriptors
+        when (updatedDescriptor.UUID.UUIDString) {
+            CBUUIDCharacteristicFormatString -> {
+                return updatedDescriptor.value as NSData
+            }
+            CBUUIDCharacteristicAggregateFormatString,
+            CBUUIDCharacteristicUserDescriptionString,
```

Comment:
For `CBUUIDCharacteristicAggregateFormatString` the apple docs do not specify what the contents of the descriptor value are. Do we know for sure this one has `String` content?

Reply:
No actually I'm not sure, I kind of trusted the  [StackOverflow answer](https://stackoverflow.com/questions/52816049/how-to-convert-cbdescriptor-value-to-string)  for that value.